### PR TITLE
enhance download experience + add context menu

### LINKED ISSUE #73: Improve download attachments experience
Using [electron-dl](https://github.com/sindresorhus/electron-dl) to give the app even more native feeling.

### PR DESCRIPTION
**Pull request by timche**
Closes #73 and #74

### CODE REVIEW

File: src/app.ts [hunk 1]
```diff
@@ -133,12 +140,14 @@ function addCustomCSS(windowElement: BrowserWindow): void {
   }
 }
 
-app.on('ready', () => {
-  createWindow()
+;(async () => {
```
Comment (markypython) [inline src/app.ts]: Why this change?

File: src/app.ts [hunk 2]
```diff
@@ -12,6 +12,8 @@ import {
 import { autoUpdater } from 'electron-updater'
 import utils, { is } from 'electron-util'
 import log from 'electron-log'
+import electronDl from 'electron-dl'
+import electronContextMenu from 'electron-context-menu'
```
Comment (markypython) [inline src/app.ts]: Does this conflict with the context menu in gmail when right clicking on an email?

File: src/app.ts (hunk 1 above)
Comment (timche) [reply src/app.ts]: Actually not really needed. I just saw in the `electron-dl` example that `app.whenReady()` can be used as a convenient alternative. I can revert this.

File: src/app.ts (hunk 2 above)
Comment (timche) [reply src/app.ts]: Can confirm that it doesn't conflict.

File: src/app.ts (hunk 1 above)
Comment (markypython) [reply src/app.ts]: Yeah, I think the `app.on('ready', () => ...)` is cleaner and fits better with the rest of the app.

Comment (timche) [reply src/app.ts]: Reverted, thanks for bringing this up.

**Comment by markypython** [comment]
I just realized that this plugin shows a badge count for downloads.  We should probably disable that since the badge is already used for unread email count.

https://www.npmjs.com/package/electron-dl#showbadge

**Comment by timche** [comment]
Good catch, fixed.
